include/sundials/sundials_context.h

```diff
@@ -19,51 +19,33 @@
 #ifndef _SUNDIALS_CONTEXT_H
 #define _SUNDIALS_CONTEXT_H
 
-#include "sundials/sundials_logger.h"
-#include "sundials/sundials_profiler.h"
-#include "sundials/sundials_types.h"
+#include <sundials/sundials_types.h>
+#include <sundials/sundials_logger.h>
+#include <sundials/sundials_profiler.h>
+#include <sundials/impl/sundials_context_impl.h>
 
-struct SUNErrHandler_;
-
-struct SUNContext_ {
-  SUNProfiler profiler;
-  sunbooleantype own_profiler;
-  SUNLogger logger;
-  sunbooleantype own_logger;
-  SUNErrCode last_err;
-  struct SUNErrHandler_ *err_handler;
-  void *comm;
-};
-
-typedef struct SUNContext_ *SUNContext;
-
-/* Shortcut macro to access the SUNContext declared with SUNDeclareContext. */
-#define SUNCTX sunctx_
-
-/* The SUNDeclareContext macro is used to declare the SUNContext 
-   object to be used a function. */
-#define SUNDeclareContext(sunctx) SUNContext SUNCTX = sunctx
+typedef struct SUNContext_* SUNContext;
 
 #ifdef __cplusplus /* wrapper to enable C++ usage */
 extern "C" {
 #endif
 
 SUNDIALS_EXPORT
-SUNErrCode SUNContext_Create(void *comm, SUNContext *ctx);
+SUNErrCode SUNContext_Create(void* comm, SUNContext* ctx);
 
 SUNDIALS_EXPORT
-SUNErrCode SUNContext_GetLastError(SUNContext sunctx, SUNErrCode *last_err);
+SUNErrCode SUNContext_GetLastError(SUNContext sunctx, SUNErrCode* last_err);
 
 SUNDIALS_EXPORT
 SUNErrCode SUNContext_PeekLastError(SUNContext sunctx, SUNErrCode* last_err);
 
 SUNDIALS_EXPORT
-struct SUNErrHandler_ *SUNContext_PushErrHandler(
-    SUNContext sunctx,
-    int (*err_fn)(int line, const char *func, const char *file, const char *msg,
-                  SUNErrCode err_code, void *err_user_data,
-                  struct SUNContext_ *sunctx),
-    void *err_user_data);
+struct SUNErrHandler_* SUNContext_PushErrHandler(
```

Comment:
It would be nice to use `SUNErrHandler` here if possible. It will probably require moving the forward declaration to a different header the header file.